dandi/pynwb_utils.py

```diff
@@ -262,9 +263,82 @@ def _get_pynwb_metadata(path: str | Path | Readable) -> dict[str, Any]:
         # get external_file data:
         out["external_file_objects"] = _get_image_series(nwb)
 
+        # Calculate session duration for metadata
+        session_duration = _get_session_duration(nwb)
+        if session_duration is not None and out.get("session_start_time") is not None:
```

Comment:
```suggestion
        # Calculate session duration for metadata
        if out.get("session_start_time") is not None \
            and ((session_duration := _get_session_duration(nwb) is not None):
```
would need to be checked but the point -- there is no point to call `_get_session_duration` at all if there is no "session_start_time".  

But it also made me wonder if it is legit at all to get duration as `max(ends) - min(starts)` and add to some start and claim that it is a start of session... Why then not just to take `max(ends)` as the end of the session?  E.g. what if session started at 0, first recording at 1, and ended at 2?

Reply:
but it is just an assumption that whatever minimal timestamp you find (besides `session_start_time`) corresponds to the same time point. Is that correct?
Did you try running this through nwb's in the archive to see what you get and whether it makes sense?